src/llama_stack/providers/utils/memory/openai_vector_store_mixin.py

```diff
@@ -819,6 +841,32 @@ async def openai_attach_file_to_vector_store(
                     message="No chunks were generated from the file",
                 )
             else:
+                # Validate embedding model and dimension are available
+                if not embedding_model:
+                    raise ValueError(f"Embedding model not found in vector store {vector_store_id} metadata")
+                if not embedding_dimension:
+                    raise ValueError(f"Embedding dimension not found in vector store {vector_store_id} metadata")
+
+                # Generate embeddings for all chunks before insertion
+                if not hasattr(self, "inference_api") or not self.inference_api:
+                    raise ValueError(
+                        "Inference API is required for embedding generation. The implementing class must set self.inference_api."
+                    )
```

Comment:
this error is going to the client, who doesn't know what an Inference API is and has no way of addressing the error. a 400 error is a client error. the client should be able to fix in the request and try again. however, there is nothing the client can do. this should be an internal server error, and ideally caught at stack startup where the admin can understand and fix it.